### PR TITLE
Fixes #4912 direct message patient update fails

### LINKED ISSUE #4912: OpenEMR received Direct messages fail when trying to update patient for message
When attempting to update a message received via Direct, the patient id of 0 is treated as a valid patient id and sent as a patient to interface/main/multi_patient_finders.php file.  The multi select shows a 'false' row which you can't delete or get rid of.  When you attempt to select a patient and hit the OK button the messages screen dies and you have to close the OpenEMR window tab completely (not the browser, the tabbed pane inside OpenEMR) in order to load the messages up again.

Looking over the code I couldn't seeing anything that has changed since @lcmaas originally put the code dealing with patient selection.  Near as I can tell it must be due to a php type error conversion where a pid integer value of 0 would type cast to an empty string '".  I will put up a PR shortly that specifically checks against this condition and fixes the patient selection update for direct messages.

I also noticed that the multi select doesn't give any error messages and fails if you don't hit the buttons in the right order.  I've added in user error messages to help with that also.

### PR DESCRIPTION
**Pull request by adunsulag**
Fixes #4912 Made it so you can update the patient for a direct message.

There's been a lot of hands in the messages.php file and it really needs to be refactored into multiple files due to how many branches and its cyclomatic complexity.  I'd greatly appreciate people taking a review of it to make sure I'm not breaking something else here.

Also added helper error messages to the multi select so you know that you need to add at list one patient to the selection before hitting the ok button.

### CODE REVIEW

File: interface/main/finder/multi_patients_finder.php [hunk 1]
```diff
@@ -231,7 +231,10 @@
 $('#add-to-list').on('click', function (e) {
     e.preventDefault();
 
-    if($('#by-name').val() == '')return;
+    if($('#by-name').val() == '') {
+        alert("<?php echo xls("You must provide a patient name or id to add to the list"); ?>");
```
Comment (bradymiller) [inline interface/main/finder/multi_patients_finder.php:235]: use xlj instead (note the outside quotes are also removed) (this function basically uses json_encode to escape in js context which adds the quotes):
```
alert(<?php echo xlj("You must provide a patient name or id to add to the list"); ?>);
```

File: interface/main/finder/multi_patients_finder.php [hunk 2]
```diff
@@ -242,7 +245,10 @@
     $.each(patientsList, function (key, patient) {
         if (patient.pid == currentResult.pid) exist = true;
     })
-    if(exist)return;
+    if(exist){
+        alert("<?php echo xls("This patient has already been added to the list"); ?>");
```
Comment (bradymiller) [inline interface/main/finder/multi_patients_finder.php:249]: use xlj and drop outside quotes

File: interface/main/finder/multi_patients_finder.php [hunk 3]
```diff
@@ -273,10 +279,15 @@ function removePatient(pid) {
 
 //send array of patients to function 'setMultiPatients' of the opener
 function selPatients() {
-    if (opener.closed || ! opener.setMultiPatients)
+    if (!(patientsList && patientsList.length)) {
+        alert("<?php echo xls("You must add a patient to the list before hitting ok"); ?>");
```
Comment (bradymiller) [inline interface/main/finder/multi_patients_finder.php:283]: use xlj and drop outside quotes

File: library/pnotes.inc [hunk 4]
```diff
@@ -495,7 +495,9 @@ function updatePnotePatient($id, $patient_id)
     $activity = $assigned_to ? '1' : '0';
 
     $pid = $row['pid'];
+
     if ($pid != 0 || (int)$patient_id < 1) {
+        (new \OpenEMR\Common\Logging\SystemLogger())->errorLogCaller("invalid operation", ['id' => $id, 'patient_id' => $patient_id, 'pid' => $pid]);
```
Comment (bradymiller) [inline library/pnotes.inc:500]: place in use block and use short form

**Comment by bradymiller** [comment]
2DV (1)

**Comment by adunsulag** [comment]
@bradymiller all the suggestions should be fixed now.  Let me know if it all looks good and I'll merge it.

**Pull request merged**